homeassistant/components/history_stats/sensor.py

```diff
@@ -319,15 +354,13 @@ def update_period(self):
         # Calculate start or end using the duration
         if start is None:
             start = end - self._duration
-        if end is None:
-            end = start + self._duration
 
         if start > dt_util.now():
             # History hasn't been written yet for this period
             return
-        if dt_util.now() < end:
-            # No point in making stats of the future
-            end = dt_util.now()
```

Comment:
If we shrink the duration to now the ratio is wrong because its being calculated based on a shorter duration.